packages/esm-patient-chart-app/src/clinical-views/utils/helpers.ts

```diff
@@ -0,0 +1,192 @@
+  return rightDate.getTime() - leftDate.getTime();
+}
+
+export function findObs(encounter: Encounter, obsConcept: string): Observation | undefined {
+  const allObs = encounter?.obs?.filter((observation) => observation.concept.uuid === obsConcept) || [];
+  return !allObs ? undefined : allObs.length === 1 ? allObs[0] : allObs.sort(obsArrayDateComparator)[0];
+}
+
+export function getObsFromEncounters(encounters: Encounter, obsConcept: Concept) {
+  const filteredEnc = encounters?.find((enc) => enc.obs.find((obs) => obs.concept.uuid === obsConcept));
+  return getObsFromEncounter(filteredEnc, obsConcept);
+}
+
+export function resolveValueUsingMappings(encounter: Encounter, concept: string, mappings) {
+  const obs = findObs(encounter, concept);
+  for (const key in mappings) {
+    if (typeof obs?.value === 'object' && 'uuid' in obs.value) {
+      if (mappings[key] === obs.value.uuid) {
+        return key;
+      }
+    }
+  }
+  return '--';
+}
+
+export function getConditionalConceptValue(encounter: any, conditionalConceptMappings, isDate) {
+  const { trueConcept, nonTrueConcept, dependantConcept, conditionalConcept } = conditionalConceptMappings;
+  const obsValue = findObs(encounter, dependantConcept)?.value;
+  const dependantUuid = typeof obsValue === 'object' && 'uuid' in obsValue ? obsValue.uuid : null;
+  const finalConcept = dependantUuid === conditionalConcept ? trueConcept : nonTrueConcept;
+  return getObsFromEncounter(encounter, finalConcept, isDate);
+}
+
+export function getConceptFromMappings(encounter: Encounter, concepts) {
+  for (const concept of concepts) {
+    const obs = findObs(encounter, concept);
+    if (obs && obs.value) {
+      return concept;
+    }
+  }
+  return null;
+}
+
+export function getMultipleObsFromEncounter(encounter: Encounter, obsConcepts: Array<string>) {
+  let observations = [];
+  obsConcepts.map((concept) => {
+    const obs = getObsFromEncounter(encounter, concept);
+    if (obs !== '--') {
+      observations.push(obs);
+    }
+  });
+
+  return observations.length ? observations.join(', ') : '--';
+}
+
+export function getObsFromEncounter(
+  encounter: Encounter,
+  obsConcept,
+  isDate?: Boolean,
+  isTrueFalseConcept?: Boolean,
+  type?: string,
+  fallbackConcepts?: Array<string>,
+  secondaryConcept?: string,
+  t?: TFunction,
+) {
+  let obs = findObs(encounter, obsConcept);
+
+  if (!encounter || !obsConcept) {
+    return '--';
+  }
+
+  // Not sure if these concepts are similar in all implementations,
+  // There might be a better way to use a generic value
+  if (isTrueFalseConcept) {
+    if (typeof obs?.value === 'object') {
+      if (
+        (obs?.value?.uuid != esmPatientChartSchema.trueConceptUuid._default && obs?.value?.name?.name !== 'Unknown') ||
+        obs?.value?.name?.name === 'FALSE'
+      ) {
+        return 'No';
+      } else if (obs?.value?.uuid == esmPatientChartSchema.trueConceptUuid._default) {
+        return 'Yes';
```

Comment:
Surely these strings need to be translatable? It would almost be better to use the actual names from the concept.